consumer-forward-service/src/main/java/com/vdobrikov/microservices/communication/rabbit/consumer/ForwardingConsumer.java

```diff
@@ -0,0 +1,57 @@
+package com.vdobrikov.microservices.communication.rabbit.consumer;
+
+import com.vdobrikov.microservices.communication.rabbit.messaging.Publisher;
+import com.vdobrikov.microservices.communication.rabbit.messaging.Subscriber;
+import com.vdobrikov.microservices.communication.rabbit.messaging.Wrapper;
+import com.vdobrikov.microservices.communication.rabbit.messaging.exception.MessagingException;
+import com.vdobrikov.microservices.communication.rabbit.messaging.rabbit.RabbitMQPublisher;
+import com.vdobrikov.microservices.communication.rabbit.messaging.rabbit.RabbitMQSubscriber;
+import io.vavr.control.Try;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+@Slf4j
+public class ForwardingConsumer {
+    private final Subscriber subscriber;
+    private final Publisher publisher;
+    private final String topic;
+    private final String host;
+
+    public ForwardingConsumer() throws IOException {
+        Properties properties = new Properties();
+        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("application.properties")) {
+            properties.load(is);
+        }
+        host = properties.getProperty("rabbitmq.host");
```

Comment:
With containerized services the easiest way to configure is to use simple environment variables as changing them does not require to re-build the docker image (as opposite to config files). More advanced option is to use Kubernetes config maps.

Reply:
I thought so, but came into conclusion that hostname is something more static in terms of one deployment so decided to store it in file. Maybe the best solution would be the hybrid one: use file if no env var.

Reply:
You can use Kubernetes Configmaps. Also the solution is to extend framework standard configuration system in such way, so that it would look at the env variables as well as into config files. See more in this article https://dzone.com/articles/configuring-spring-boot-on-kubernetes-with-configm